spk/transmission/src/service-setup.sh

```diff
@@ -14,119 +14,55 @@ TRANSMISSION="${SYNOPKG_PKGDEST}/bin/transmission-daemon"
-            NEW_INCOMPLETE_DIR="${wizard_volume:=/volume1}/${wizard_download_dir:=/downloads}/incomplete"
-            NEW_WATCH_DIR="${wizard_volume:=/volume1}/${wizard_download_dir:=/downloads}/watch"
-            mkdir -p "${NEW_INCOMPLETE_DIR}"
-            mkdir -p "${NEW_WATCH_DIR}"
-
-            # update folders in config file according to the wizard
-            SETTINGS=$(cat "${CFG_FILE}")
-            SETTINGS=$(echo "$SETTINGS" | jq '."watch-dir-enabled"=true | ."incomplete-dir-enabled"=true')
-            SETTINGS=$(echo "$SETTINGS" | jq --arg path ${NEW_DOWNLOAD_DIR} '."download-dir"=$path')
-            SETTINGS=$(echo "$SETTINGS" | jq --arg path ${NEW_INCOMPLETE_DIR} '."incomplete-dir"=$path')
-            SETTINGS=$(echo "$SETTINGS" | jq --arg path ${NEW_WATCH_DIR} '."watch-dir"=$path')
-            echo "${SETTINGS}" > ${CFG_FILE}
-
-            # move files when folders are changed
-            shopt -s dotglob # move hidden folder/files too
-            if [ -n "${DOWNLOAD_DIR}" ] && [ $(realpath "${DOWNLOAD_DIR}") != $(realpath "${NEW_DOWNLOAD_DIR}") ]; then
-                # move only files from previous download folder that are created by this package (${EFF_USER})
-                find "${DOWNLOAD_DIR}" -maxdepth 1 -user ${EFF_USER} -exec mv -nv {} "${NEW_DOWNLOAD_DIR}/" \;
-            fi
-            if [ -n "${INCOMPLETE_DIR}" ] && [ $(realpath "${INCOMPLETE_DIR}") != $(realpath "${NEW_INCOMPLETE_DIR}") ]; then
-                mv -nv "${INCOMPLETE_DIR}/*" "${NEW_INCOMPLETE_DIR}/"
-            fi
-            if [ -n "${WATCH_DIR}" ] && [ $(realpath "${WATCH_DIR}") != $(realpath "${NEW_WATCH_DIR}") ]; then
-                mv -nv "${WATCH_DIR}/*" "${NEW_WATCH_DIR}/"
+        # Create the managed folders
+        for item in "${TXN_FOLDRS[@]}"; do
+            folder="$TXN_DNLOAD/$item"
+            if [ "${SYNOPKG_DSM_VERSION_MAJOR}" -ge 7 ]; then
+                mkdir -p "$folder"
+            else
+                /bin/su "${EFF_USER}" -s /bin/sh -c "mkdir -p $folder"
+                set_syno_permissions "$folder" "${GROUP}"
             fi
-            shopt -u dotglob
+            TXN_PATHS+=("$folder")
+        done
 
-        else
-
-            # Apply permissions
-            if [ -n "${DOWNLOAD_DIR}" ] && [ -d "${DOWNLOAD_DIR}" ]; then
-                set_syno_permissions "${DOWNLOAD_DIR}" "${GROUP}"
-            fi
-            if [ -n "${INCOMPLETE_DIR}" ] && [ -d "${INCOMPLETE_DIR}" ]; then
-                set_syno_permissions "${INCOMPLETE_DIR}" "${GROUP}"
-            fi
-            if [ -n "${WATCH_DIR}" ] && [ -d "${WATCH_DIR}" ]; then
-                set_syno_permissions "${WATCH_DIR}" "${GROUP}"
+        # Edit the configuration according to the wizard
+        sed -e "s|@username@|${wizard_username:=admin}|g" \
+            -e "s|@password@|${wizard_password:=admin}|g" \
+            -i "${CFG_FILE}"
+        i=0
+        while [ $i -lt ${#TXN_FUNCTS[@]} ]; do
```

Comment:
The difference in the iteration is to keep the folders and the functions separate. Specifically, in the configuration file there is a line for the watch folder for `watch_dir_enabled` which can only work if we reference the function name array rather than the folder name array. Keeping the use of the variables separate for me was a way to avoid errors (i.e. the first loop only references `TXN_FOLDRS` where as the second loop only references `TXN_FUNCTS` and the two are not mixed).

Reply:
I meant using the same kind of loop actually (e.g. here you are using a cursor based loop while for the folders one you used a "for each" loop).

Reply:
Oh! No specific reason. I supposed I could change the first loop to be a cursor based loop as well and change the line:

```
folder="$TXN_DNLOAD/$item"
```

to

```
folder="$TXN_DNLOAD/${TXN_FOLDRS[$i]}"
```

Doesn't really matter to me.